java-checks/src/main/java/org/sonar/java/checks/unused/UnusedMethodParameterCheck.java

```diff
@@ -25,25 +25,34 @@
 import org.sonar.check.Rule;
 import org.sonar.java.matcher.MethodMatcher;
 import org.sonar.java.matcher.MethodMatcherCollection;
+import org.sonar.java.model.JavaTree;
 import org.sonar.java.model.ModifiersUtils;
 import org.sonar.java.model.declaration.MethodTreeImpl;
 import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
 import org.sonar.plugins.java.api.JavaFileScannerContext;
 import org.sonar.plugins.java.api.semantic.Symbol;
+import org.sonar.plugins.java.api.tree.AnnotationTree;
 import org.sonar.plugins.java.api.tree.BlockTree;
 import org.sonar.plugins.java.api.tree.IdentifierTree;
+import org.sonar.plugins.java.api.tree.ListTree;
+import org.sonar.plugins.java.api.tree.LiteralTree;
 import org.sonar.plugins.java.api.tree.MethodTree;
 import org.sonar.plugins.java.api.tree.Modifier;
 import org.sonar.plugins.java.api.tree.Tree;
 import org.sonar.plugins.java.api.tree.VariableTree;
 
 import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
 import java.util.List;
 
 @Rule(key = "S1172")
 public class UnusedMethodParameterCheck extends IssuableSubscriptionVisitor {
 
   private static final String AUTHORIZED_ANNOTATION = "javax.enterprise.event.Observes";
+  private static final String SUPPRESS_WARNINGS_ANNOTATION = "java.lang.SuppressWarnings";
+  private static final Collection<String> EXCLUDED_WARNINGS_SUPPRESIONS = Collections.unmodifiableCollection(Arrays.asList("\"rawtypes\"", "\"unchecked\""));
```

Comment:
As we are already using it here : https://github.com/SonarSource/sonar-java/pull/917/files#diff-f8c834c0aca82abca36725beddf6ab85R62 I think using ImmutableList.of would be a better choice in this case.